File: web/src/main/angular/app/model/PatientDataModel.ts
import {Condition, Encounter, MedicationRequest, Observation, Procedure, ServiceRequest} from "./fhir";

export class PatientDataModel {
    public conditions: Condition[];
    public encounters: Encounter[];
    public medicationRequests: MedicationRequest[];
    public procedures: Procedure[];
    public observations: Observation[];
    public serviceRequests: ServiceRequest[];
}

File: core/src/main/java/com/lantanagroup/link/model/PatientDataModel.java
package com.lantanagroup.link.model;

import lombok.Getter;
import lombok.Setter;
import org.hl7.fhir.r4.model.*;

import java.util.List;

@Getter @Setter
public class PatientDataModel {
    List<Condition> conditions;
    List<MedicationRequest> medicationRequests;
    List<Procedure> procedures;
    List<Encounter> encounters;
    List<Observation> observations;
    List<ServiceRequest> serviceRequests;
}
